src/parser/Parser.ts

```diff
@@ -387,8 +387,10 @@ export class Parser {
                     overrideKeyword = this.advance();
                 }
 
+                let fieldAnnotations = this.pendingAnnotations;
```

Comment:
Why do you grab the annotations here, but then clear them inside the if statement 2 lines below? I would expect the variable and the clear to happen at the same place.

Reply:
we need to do this or block annotations will carry over onto class members